src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+    let nextState;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        nextState = {
+          ...currentState, ...extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        nextState = { ...currentState };
+        keysToRemove.forEach((key) => delete nextState[key]);
+        break;
+
+      case 'clear':
+        nextState = {};
+        break;
+    }
+
+    result.push(nextState);
```

Comment:
Consider renaming 'nextState' to a more descriptive name, e.g., 'updatedState'. This will make the code easier to understand for other developers.